streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -17,116 +17,109 @@
 
 package org.apache.kafka.streams.kstream;
 
+import org.apache.kafka.common.annotation.InterfaceStability;
 import org.apache.kafka.common.serialization.Serde;
 import org.apache.kafka.streams.KeyValue;
 import org.apache.kafka.streams.processor.ProcessorSupplier;
 
 /**
- * KStream is an abstraction of an event stream in key-value pairs.
+ * KStream is an abstraction of a <i>record stream</i> in key-value pairs.
  *
  * @param <K> Type of keys
  * @param <V> Type of values
  */
+@InterfaceStability.Unstable
 public interface KStream<K, V> {
 
     /**
-     * Creates a new instance of KStream consists of all elements of this stream which satisfy a predicate
+     * Create a new instance of {@link KStream} consists of all elements of this stream which satisfy a predicate.
      *
-     * @param predicate the instance of Predicate
-     * @return the instance of KStream with only those elements that satisfy the predicate
+     * @param predicate     the instance of {@link Predicate}
      */
     KStream<K, V> filter(Predicate<K, V> predicate);
 
     /**
-     * Creates a new instance of KStream consists all elements of this stream which do not satisfy a predicate
+     * Create a new instance of KStream consists all elements of this stream which do not satisfy a predicate.
      *
-     * @param predicate the instance of Predicate
-     * @return the instance of KStream with only those elements that do not satisfy the predicate
+     * @param predicate     the instance of {@link Predicate}
      */
     KStream<K, V> filterOut(Predicate<K, V> predicate);
 
     /**
-     * Creates a new instance of KStream by applying transforming each element in this stream into a different element in the new stream.
+     * Create a new instance of {@link KStream} by transforming each element in this stream into a different element in the new stream.
      *
-     * @param mapper the instance of KeyValueMapper
-     * @param <K1>   the key type of the new stream
-     * @param <V1>   the value type of the new stream
-     * @return the instance of KStream
+     * @param mapper        the instance of {@link KeyValueMapper}
+     * @param <K1>          the key type of the new stream
+     * @param <V1>          the value type of the new stream
      */
     <K1, V1> KStream<K1, V1> map(KeyValueMapper<K, V, KeyValue<K1, V1>> mapper);
 
     /**
-     * Creates a new instance of KStream by transforming each value in this stream into a different value in the new stream.
+     * Create a new instance of {@link KStream} by transforming each value in this stream into a different value in the new stream.
```

Comment:
Perhaps clarify to: "by transforming the value of each element in this stream into a different value in the new stream"